content/intro-to-reading-data.md

```diff
@@ -101,90 +103,95 @@ The balance of the account at CenYq6bDRB7p73EjsPEpiYN7uveyPUTdXkDkgUduboaN is 0.
+Make a new file called `check-balance.ts`, substituting your public key in for `<your public key>`.
 
-### 1. Get oriented
+The script loads the public key, connects to DevNet, and checks the balance:
 
-Once you’ve got the starter code, take a look around. Install the dependencies with `npm install` and then run the app with `npm run dev`. Notice that no matter what you put into the address field, when you click “Check SOL Balance” the balance will be a placeholder value of 1000.
+```
+import { Connection, LAMPORTS_PER_SOL, PublicKey } from "@solana/web3.js";
 
-Structurally, the app is composed of `index.tsx` and `AddressForm.tsx`. When a user submits the form, the `addressSubmittedHandler` in `index.tsx` gets called. That’s where we’ll be adding the logic to update the rest of the UI.
+const publicKey = new PublicKey("<your public key>");
 
-### 2. Install dependencies
+const connection = new Connection("https://api.devnet.solana.com", "confirmed");
 
-Use `npm install @solana/web3.js` to install our dependency on Solana’s web3 library.
+const balanceInLamports = await connection.getBalance(publicKey);
 
-### 3. Set the address balance
+const balanceInSOL = balanceInLamports / LAMPORTS_PER_SOL;
 
-First, import `@solana/web3.js` at the top of `index.tsx`.
+console.log(
+  `💰 Finished! The balance for the wallet at address ${publicKey} is ${balanceInSOL}!`
+);
 
-Now that the library is available, let’s go into the `addressSubmittedHandler()` and create an instance of `PublicKey` using the address value from the form input. Next, create an instance of `Connection` and use it to call `getBalance()`. Pass in the value of the public key you just created. Finally, call `setBalance()`, passing in the result from `getBalance`. If you’re up to it, try this independently instead of copying from the code snippet below.
+```
 
-```typescript
-import type { NextPage } from 'next'
-import { useState } from 'react'
-import styles from '../styles/Home.module.css'
-import AddressForm from '../components/AddressForm'
-import * as web3 from '@solana/web3.js'
-
-const Home: NextPage = () => {
-  const [balance, setBalance] = useState(0)
-  const [address, setAddress] = useState('')
-
-  const addressSubmittedHandler = async (address: string) => {
-    setAddress(address)
-    const key = new web3.PublicKey(address)
-    const connection = new web3.Connection(web3.clusterApiUrl('devnet'));
-    const balance = await connection.getBalance(key);
-    setBalance(balance / web3.LAMPORTS_PER_SOL);
-  }
-  ...
-}
+Save this to a file, and `npx esrun check-balance.ts`. You should something like:
```

Comment:
I'm only just now realizing that you're using `esrun`. Haven't used it before. We use ts-node throughout the rest of the course (though it's typically inside a scripts command in the package.json). Is there a reason to use `esrun` over `ts-node`?

Reply:
cc @mikemaccana